### ISSUE TITLE
Remove build dependency on golang

### ISSUE THREAD

**Issue by codefromthecrypt** [envoyproxy/envoy#17886]
*Title*: Remove build dependency on golang

*Description*:
While marketed as always compatible, recent versions of go break especially around how modules work. Rather than embed this problem into Envoy, causing other tensions like go's platform matrix, could we not depend on go? At least for release builds?

These are the files I can see that reference go source. There may be some incidental dependencies on go binaries I'm not aware of, but perhaps could be handled differently or isolated from a release.
```
./generated/docs/_downloads/ef307e1c6786b51ee38110d4bc3a839b/resource.go
./examples/dynamic-config-cp/resource.go
./examples/ext_authz/auth/grpc-service/main.go
./examples/ext_authz/auth/grpc-service/pkg/auth/users.go
./examples/ext_authz/auth/grpc-service/pkg/auth/v3/auth.go
./examples/load-reporting-service/server/lrs_server.go
./examples/load-reporting-service/main.go
./examples/grpc-bridge/server/service.go
./api/test/build/go_build_test.go
```

*Relevant Links*:

https://github.com/Homebrew/homebrew-core/pull/83413 - go 1.17 upgrade of homebrew. Even if it is a red herring, even having someone validate if envoy build works or not costs them hours for the build to complete or fail. Removing the need to think about go, would save a lot of ecosystem time.

**Comment by codefromthecrypt** [comment]
@moderation the below did happen and it did say problem downloading go, and it only happened on arm64

```
==> /opt/homebrew/opt/bazelisk/bin/bazelisk build --compilation_mode=opt --curses=no --show_task_finish 
Last 15 lines from /Users/thebots/Library/Logs/Homebrew/envoy/01.bazelisk:
ERROR: An error occurred during the fetch of repository 'go_sdk':
   Traceback (most recent call last):
	File "/private/tmp/envoy-20210827-59011-735pgh/.brew_home/_bazel/6734651a1108d63ae9e743df04c9a20c/external/io_bazel_rules_go/go/private/sdk.bzl", line 98, column 13, in _go_download_sdk_impl
		fail("unsupported platform {}".format(platform))
Error in fail: unsupported platform darwin_arm64
ERROR: Error fetching repository: Traceback (most recent call last):
	File "/private/tmp/envoy-20210827-59011-735pgh/.brew_home/_bazel/6734651a1108d63ae9e743df04c9a20c/external/io_bazel_rules_go/go/private/sdk.bzl", line 98, column 13, in _go_download_sdk_impl
		fail("unsupported platform {}".format(platform))
Error in fail: unsupported platform darwin_arm64
ERROR: Analysis of target '//source/exe:envoy-static' failed; build aborted: unsupported platform darwin_arm64
INFO: Elapsed time: 83.843s
INFO: 0 processes.
FAILED: Build did NOT complete successfully (18 packages loaded, 7 targets configured)
FAILED: Build did NOT complete successfully (18 packages loaded, 7 targets configured)
```

Whether or not you (an expert) are able to have a perfect working environment isn't the motivation of this issue. It is if the distraction caused by go and there have been, not just this one, is a required distraction or an optional one.

I can already tell you are in favor of keeping it, but let's not pretend everyone else's experiences and time loss never happened

**Comment by codefromthecrypt** [comment]
PS it is a good point to try to fix homebrew  #17500 hopefully can switch things to ad-hoc after the fact fixes to proactive part of a release. You'd get less non-experts commenting issues like this if the project team owned all their releases.

**Comment by mattklein123** [comment]
I'm going to close this out as this is not going to realistically happen.

> You'd get less non-experts commenting issues like this if the project team owned all their releases.

In a world of infinite resources this sounds great. We don't live in such a world. Help appreciated.
